skimage/restoration/inpaint.py

```diff
@@ -0,0 +1,134 @@
+from __future__ import print_function, division
+
+import numpy as np
+import skimage
+from scipy import sparse
+from scipy.sparse.linalg import spsolve
+from scipy.ndimage.filters import laplace
+
+
+def inpaint_biharmonic(img, mask, multichannel=False):
+    """Inpaint masked points in image with biharmonic equations.
+
+    Parameters
+    ----------
+    img : nD{+color channel} np.ndarray
+        Input image.
+    mask : nD np.ndarray
+        Array of pixels to be inpainted. Have to be the same size as one
+        of the 'img' channels. Unknown pixels has to be represented with 1, 
+        known pixels - with 0.
+    multichannel : boolean, optional
+        If True, the last `img` dimension is considered as a color channel.
+
+    Returns
+    -------
+    out : nD{+color channel} np.array
+        Input image with masked pixels inpainted.
+
+    Example
+    -------
+    >>> img = np.tile(np.square(np.linspace(0, 1, 5)), (5, 1))
+    >>> mask = np.zeros_like(img)
+    >>> mask[2, 2:] = 1
+    >>> mask[1, 3:] = 1
+    >>> mask[0, 4:] = 1
+    >>> out = inpaint_biharmonic(img, mask)
+
+    References
+    ----------
+    Algorithm is based on:
+    .. [1]  N.S.Hoang, S.B.Damelin, "On surface completion and image inpainting
+            by biharmonic functions: numerical aspects",
+            http://www.ima.umn.edu/~damelin/biharmonic
+    """
+    
+    def _inpaint(img, mask):
+        out = np.copy(img)
+
+        # Initialize sparse matrices
+        matrix_unknown = sparse.lil_matrix((np.sum(mask), out.size))
+        matrix_known = sparse.lil_matrix((np.sum(mask), out.size))
+
+        def _get_neighborhood(idx, radii):
```

Comment:
Also, why radii instead of radius? You only use it with a single radius, so might as well make the function simple? The function call will be more readable with "radius=2" than with "radii=np.array([2])".